web/src/components/SharingPopup.tsx

```diff
@@ -0,0 +1,258 @@
+import Tooltip from './Tooltip'
+
+type SharingPopupProps = {
+  shareUrl: string
+  title: string
+  flow: 'vertical' | 'horizontal'
+  direction: UiDirectionType
+  route: RouteType
+}
+
+const TooltipContainer = styled.div<{
+  flow: 'vertical' | 'horizontal'
+  active: boolean
+  direction: UiDirectionType
+  additionalPadding: number
+  horizontalPosition?: number
+}>`
+  background-color: ${props => props.theme.colors.backgroundColor};
+  padding: 8px;
+  border: 2px solid ${props => props.theme.colors.textDecorationColor};
+  width: max-content;
+  position: absolute;
+  display: flex;
+  z-index: 2000;
+  opacity: 0;
+
+  ${props =>
+    props.flow === 'vertical' &&
+    css`
+      flex-flow: column-reverse;
+      transform: translateY(-100%);
+    `};
+
+  ${props =>
+    props.flow === 'horizontal' &&
+    (props.direction === 'ltr'
+      ? css`
+          transform: translateX(30%);
+        `
+      : css`
+          transform: translate(-30%);
+        `)};
+
+  ${props =>
+    props.active &&
+    css`
+      animation: tooltips 300ms ease-out forwards;
+    `};
+
+  &:before,
+  &:after {
+    position: absolute;
+    content: '';
+    display: block;
+  }
+
+  &:before {
+    z-index: 2000;
+    border-bottom: 10px solid ${props => props.theme.colors.backgroundColor};
+    border-left: 10px solid transparent;
+    border-right: 10px solid transparent;
+
+    ${props =>
+      props.flow === 'vertical' &&
+      (props.direction === 'ltr'
+        ? css`
+            left: 20px;
+            bottom: -8px;
+            transform: rotate(-180deg);
+          `
+        : css`
+            bottom: -8px;
+            transform: translateX(-55%) rotate(180deg);
+          `)};
+
+    ${props =>
+      props.flow === 'horizontal' &&
+      (props.direction === 'ltr'
+        ? css`
+            left: -15px;
+            transform: rotate(-90deg);
+            top: 45%;
+          `
+        : css`
+            right: -15px;
+            transform: rotate(90deg);
+            top: 45%;
+          `)};
+
+    ${props =>
+      props.active &&
+      css`
+        animation: tooltips 300ms ease-out forwards;
+      `};
+  }
+
+  &:after {
+    z-index: 1000;
+    border-bottom: 11px solid ${props => props.theme.colors.textDecorationColor};
+    border-left: 11px solid transparent;
+    border-right: 11px solid transparent;
+
+    ${props =>
+      props.flow === 'vertical' &&
+      (props.direction === 'ltr'
+        ? css`
+            left: 20px;
+            bottom: -11px;
+            transform: rotate(-180deg);
+          `
+        : css`
+            bottom: -11px;
+            transform: translateX(-45%) rotate(180deg);
+          `)};
+
+    ${props =>
+      props.flow === 'horizontal' &&
+      (props.direction === 'ltr'
+        ? css`
+            left: -17px;
+            transform: rotate(-90deg) scaleX(-1);
+            top: 45%;
+          `
+        : css`
+            right: -18px;
```

Comment:
Why 18px here and 17 above? Could we in general use some more values that don't feel that random? Like two potencies?

Reply:
adjusted both to 18. you have to move the before and after icon before or after the element